packages/terra-hookshot/src/Hookshot.jsx

```diff
@@ -0,0 +1,335 @@
+import HookshotUtils from './_HookshotUtils';
+
+const ATTACHMENT_POSITIONS = [
+  'top left',
+  'top center',
+  'top right',
+  'middle left',
+  'middle center',
+  'middle right',
+  'bottom left',
+  'bottom center',
+  'bottom right',
+];
+
+const ATTACHMENT_BEHAVIORS = [
+  'auto',
+  'flip',
+  'none',
+];
+
+const propTypes = {
+  /**
+   * How the content should be positioned when the primary attachment is not available.
+   * Valid values: 'auto', 'flip', and 'none'.
+   */
+  attachmentBehavior: PropTypes.oneOf(ATTACHMENT_BEHAVIORS),
+  /**
+   * Value in px of the margin to place between the target and the content.
+   */
+  attachmentMargin: PropTypes.number,
+  /**
+   * Reference to the bounding container, will use window if nothing is provided.
+   */
+  boundingRef: PropTypes.func,
+  /**
+   * The HookshotContent to be attached.
+   */
+  children: PropTypes.element.isRequired,
+  /**
+   * String pair separated by a space using values of top, middle, bottom, and left, center, right. (ex: 'top middle')
+   */
+  contentAttachment: PropTypes.oneOf(ATTACHMENT_POSITIONS).isRequired,
+  /**
+   * String pair of top and left offset, ie "10px -4px".
+   */
+  contentOffset: PropTypes.string,
+  /**
+   * Determines whether the content should be actively positioned via hookshot.
+   */
+  isEnabled: PropTypes.bool,
+  /**
+   * Should the content be presented.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Callback function when the content has been positioned.
+   */
+  onPosition: PropTypes.func,
+  /**
+   * Required element that the content will hookshot to.
+   */
+  targetRef: PropTypes.func.isRequired,
+  /**
+   * String pair of top, middle, bottom, and left, center, right. Will mirror content if none provided. (ex: 'bottom middle')
+   */
+  targetAttachment: PropTypes.oneOf(ATTACHMENT_POSITIONS),
+  /**
+   * String pair of top and left offset, ie "10px -4px".
+   */
+  targetOffset: PropTypes.string,
+};
+
+const defaultProps = {
+  attachmentMargin: 0,
+  attachmentBehavior: 'auto',
+  contentOffset: '0 0',
+  isEnabled: false,
+  isOpen: false,
+  targetOffset: '0 0',
+};
+
+class Hookshot extends React.Component {
+  static now() {
+    return performance.now();
+  }
+
+  constructor(props) {
+    super(props);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.getNodeRects = this.getNodeRects.bind(this);
+    this.update = this.update.bind(this);
+    this.tick = this.tick.bind(this);
+    this.state = { isEnabled: props.isEnabled && props.isOpen };
+    this.listenersAdded = false;
+    this.lastCall = null;
+    this.lastDuration = null;
+    this.pendingTimeout = null;
+  }
+
+  componentDidMount() {
+    if (this.state.isEnabled) {
+      if (!this.listenersAdded) {
+        this.enableListeners();
+      }
+      this.update();
+    }
+  }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.isEnabled !== this.props.isEnabled || newProps.isOpen !== this.props.isOpen) {
```

Comment:
Since you're not implementing `shouldComponentUpdate`, you can probably ditch the prop checks and just update the state every time. Render is going to get called regardless.